libminifi/src/Processor.cpp

```diff
@@ -206,6 +207,29 @@ bool Processor::getProperty(std::string name, std::string &value)
 	}
 }
 
+bool Processor::getDynamicProperty(std::string name, std::string &value)
+{
+	if (isRunning())
+		// Because set property only allowed in non running state, we need to obtain lock avoid rack condition
+		_mtx.lock();
```

Comment:
Hi, any particular reason you did not use a lock guard ? http://en.cppreference.com/w/cpp/thread/lock_guard 
It's scoped on the stack pop so you don't need to unlock.